### ISSUE TITLE
Microsoft bot framework - Bot channel Registration. Unable to save the recorded video from skype to Azure storage account

### ISSUE THREAD

**Issue by rajeshnagendra** [microsoft/botframework-sdk#5710]
For the Microsoft Bot framework chatbot application that I am working on, I have configured the "Bot Channel Registration" and have hosted it on Azure.

One of the scenerio expects the user to record a video on skype and send it as an answer. I have an Azure function that saves the recorded video from skype to the Azure Storage account.

The issue I am encountering is, When i record a video on skype, i get to see a skype video URL in below format.

> https://smba.trafficmanager.net/apis/v3/attachments/0-sa-d2-1aadf7d75d0f15ecc97c58246cdb50fd/views/original

To gain access to the uploaded video from skype, I am providing appropriate bearer token along with the above mentioned URL but failing to get access to it.

Though the file that is uploaded from skype to the Queue (Azure function Queue triggers), the accessibility to this file is denied.

Below is the exception message:

> An error occurred while sending the request. Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..

_In short, when my chatbot is configured to a skype channel on azure, and when I record a skype video and upload the video, it should save in the Azure storage account._

Assuming the latest patches would help, I updated all the references to .NET core 3.0.1 as on today. Looking forward for a desired approach to resolve this.

--Rajesh

**Comment by v-kydela** [comment]
@rajeshnagendra - The Skype channel configuration contains the following message:

> As of October 31, 2019 the Skype channel no longer accepts new Bot publishing requests. This means that you can continue to develop bots using the Skype channel, but your bot will be limited to 100 users. You will not be able to publish your bot to a larger audience. Current Skype bots will continue to run uninterrupted. [Learn more](https://support.skype.com/en/faq/fa12091/why-are-some-features-not-available-in-skype-anymore)

Regardless, I need to understand whether your problem has anything to do with the Bot Framework. You say:

>I have an Azure function that saves the recorded video from skype to the Azure Storage account.

Then you say:

> Though the file that is uploaded from skype to the Queue (Azure function Queue triggers), the accessibility to this file is denied.

Then you say:

> _In short, when my chatbot is configured to a skype channel on azure, and when I record a skype video and upload the video, it should save in the Azure storage account._

It's unclear what you're saying the problem is. Is the video getting uploaded to Azure storage or not? Are you unable to access the file yourself or is it only inaccessible to the bot? Since this appears to be something you're doing with Skype, Azure Store, and Azure Functions, why do you think this has anything to do with the Bot Framework?

**Comment by v-kydela** [comment]
In the future, please make sure you link to all relevant posts: https://stackoverflow.com/questions/59676756/microsoft-bot-framework-bot-channel-registration-unable-to-save-the-recorded

**Comment by rajeshnagendra** [comment]
Hi Kydela,

Thank you  for responding. 

As I am building a bot using Microsoft bot framework, I have mentioned it here. 

The bot app is configured to access via skype (with bot channel registration on azure). 
While the similar scenario on webchat is allowing to upload video successfully, its failing in skype.
The issue here is that, after the file enters the queue, it doesn't succeed to upload, instead is moved to queue poison. 

I see this forum would be accessed from developers working on such scenarios. Please assist.

**Comment by rajeshnagendra** [comment]
> In the future, please make sure you link to all relevant posts: https://stackoverflow.com/questions/59676756/microsoft-bot-framework-bot-channel-registration-unable-to-save-the-recorded

Adding another link

https://social.msdn.microsoft.com/Forums/en-US/18badabe-2f3c-4988-83a7-8d1ed6c8c6ee/bot-channel-registration-unable-to-save-the-recorded-video-from-skype-to-azure-storage-account?forum=csharpgeneral

**Comment by v-kydela** [comment]
@rajeshnagendra - I still don't understand what you're doing or why, or what's going wrong. You're saying you can upload a video in Web Chat? You mean you're just attaching a prerecorded video, right? That's different from Skype, where I presume you're using the video message button or something:

![image](https://user-images.githubusercontent.com/41968495/72279927-0aebc900-35ec-11ea-90d8-987dcb5227cb.png)

Please include relevant code samples and screenshots to demonstrate what you're trying to do.

**Comment by rajeshnagendra** [comment]
Hi Kydela,

I will follow up on this issue on Microsoft Azure forum. As this  forum is dedicated to bot framework, I prefer closing this issue here. In case of any queries, I shall get back to you. Thank you very much for your precious time.

**Comment by rajeshnagendra** [comment]
> @rajeshnagendra - I still don't understand what you're doing or why, or what's going wrong. You're saying you can upload a video in Web Chat? You mean you're just attaching a prerecorded video, right? That's different from Skype, where I presume you're using the video message button or something:
> 
> ![image](https://user-images.githubusercontent.com/41968495/72279927-0aebc900-35ec-11ea-90d8-987dcb5227cb.png)
> 
> Please include relevant code samples and screenshots to demonstrate what you're trying to do.

Your understanding here is correct. I am recording a video on skype's video recording option.
